diff --git a/openapi/CMakeLists.txt b/openapi/CMakeLists.txt
index 23269de..95490e5 100644
--- a/openapi/CMakeLists.txt
+++ b/openapi/CMakeLists.txt
@@ -11,6 +11,7 @@ set(EXTERNAL_INSTALL_LOCATION ${CMAKE_CURRENT_SOURCE_DIR}/external)
 ExternalProject_Add(PISTACHE
     GIT_REPOSITORY https://github.com/pistacheio/pistache.git
     BUILD_IN_SOURCE true
+    CMAKE_ARGS -DPISTACHE_BUILD_TESTS=OFF
     INSTALL_COMMAND meson setup build --prefix=${EXTERNAL_INSTALL_LOCATION} --libdir=lib && meson install -C build
 )
 
