diff --git a/openapi/CMakeLists.txt b/openapi/CMakeLists.txt
index 95490e5..a88a4df 100644
--- a/openapi/CMakeLists.txt
+++ b/openapi/CMakeLists.txt
@@ -20,6 +20,18 @@ ExternalProject_Add(NLOHMANN
     CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${EXTERNAL_INSTALL_LOCATION} -DJSON_BuildTests=OFF
 )
 
+ExternalProject_Add(LIBWEBSOCKETS
+    GIT_REPOSITORY https://github.com/warmcat/libwebsockets.git
+    GIT_TAG v4.3-stable  # Optionally specify a version or tag
+
+    CMAKE_ARGS
+        -DCMAKE_INSTALL_PREFIX=${EXTERNAL_INSTALL_LOCATION}
+        -DLWS_WITH_SHARED=OFF
+        -DLWS_WTIH_STATIC=ON
+        -DLWS_WITHOUT_TESTAPPS=ON
+        -DLWS_WITH_SSL=OFF
+)
+
 include_directories(${EXTERNAL_INSTALL_LOCATION}/include)
 link_directories(${EXTERNAL_INSTALL_LOCATION}/lib)
 
@@ -35,5 +47,5 @@ file(GLOB SRCS
 )
 
 add_executable(${PROJECT_NAME} ${SRCS} )
-add_dependencies(${PROJECT_NAME} PISTACHE NLOHMANN)
-target_link_libraries(${PROJECT_NAME} pistache pthread)
+add_dependencies(${PROJECT_NAME} PISTACHE NLOHMANN LIBWEBSOCKETS)
+target_link_libraries(${PROJECT_NAME} websockets pistache pthread)
